content/docs/faq/webhook.md

```diff
@@ -1,12 +1,879 @@
+
+```text
+curl: (7) Failed to connect to 127.0.0.1 port 6080 after 0 ms: Connection refused ❌
+```
+
+At this point, verify that the readiness endpoint is configured on that
+same port. Let us see the logs to check that our webhook is listening on 6080
+for its readiness endpoint:
+
+```console
+$ kubectl logs -n cert-manager -l app.kubernetes.io/name=webhook | head -10
+I0607 webhook.go:129] "msg"="using dynamic certificate generating using CA stored in Secret resource"
+I0607 server.go:133] "msg"="listening for insecure healthz connections"  "address"=":6081" ❌
+I0607 server.go:197] "msg"="listening for secure connections"  "address"=":10250"
+I0607 dynamic_source.go:267] "msg"="Updated serving TLS certificate"
+...
+```
+
+In the above example, the issue was a misconfiguration of the readiness port. In
+the webhook deployment, the argument `--healthz-port=6081` was mismatched with
+the readiness configuration.
+
+## Error 2: `i/o timeout`
+
+> This error message was reported 26 times on Slack. To list these messages, do a search with `in:#cert-manager in:#cert-manager-dev "443: i/o timeout"`. The error message was reported X times in GitHub issues ([#2811](https://github.com/cert-manager/cert-manager/issues/2811 "i/o timeout from apiserver when connecting to webhook on k3s"), [#4073](https://github.com/cert-manager/cert-manager/issues/4073 "Internal error occurred: failed calling webhook"))
+
+```text
+Error from server (InternalError): error when creating "STDIN": Internal error occurred:
+  failed calling webhook "webhook.cert-manager.io": failed to call webhook:
+    Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s":
+      dial tcp 10.0.0.69:443: i/o timeout
+```
+
+When the apiserver tries to talk to the cert-manager webhook, the `SYN` packet
+is never answered, and the connection times out. If we were to run tcpdump inside
+the webhook's net namespace, we would see:
+
+```text
+192.168.1.43 (apiserver) -> 10.0.0.69 (webhook pod) TCP 44772 → 443 [SYN]
+192.168.1.43 (apiserver) -> 10.0.0.69 (webhook pod) TCP [TCP Retransmission] 44772 → 443 [SYN]
+192.168.1.43 (apiserver) -> 10.0.0.69 (webhook pod) TCP [TCP Retransmission] 44772 → 443 [SYN]
+192.168.1.43 (apiserver) -> 10.0.0.69 (webhook pod) TCP [TCP Retransmission] 44772 → 443 [SYN]
+```
+
+This issue is caused by the `SYN` packet being dropped somewhere.
+
+If you are using a "private" GKE cluster, your worker nodes won't have
+external IPs assigned. Unlike public GKE clusters where the control plane
+can freely talk to pods over any TCP port thanks to the external IPs, the
+control plane in private GKE clusters can only talk to the pods in worker
+nodes over TCP port 10250 and 443. That port is the port inside the pod,
+not the service port: the service used to expose the cert-manager webhook
+must use the `targetPort` 10250 or 443. Fortunately, by default, the
+cert-manager Helm chart uses 443 as the target port:
```

Comment:
That whole section was very confusing and full of mistakes. I have re-wrote some parts and fixed the mistakes. It should read much better; I also explained why the Service `port` doesn't matter (because kube-proxy on the control plane node transforms 443 into 10250 before it ever leaves the node)